### PR TITLE
Anubhav: Explore Component of HomePage implemented

### PR DESCRIPTION
**Pull request by AnubhavChakrabortynits**
commit-1: Made The Explore Component

commit-2: Explore Component Made Responsive for all breakPoints 

commit-3: Changing Text Animation Added in the title

commit-4 Minor glitches fixed in the animation as well as the animation duration was increased..

@codadept @resyfer

### CODE REVIEW

File: src/Components/Explore/Explore.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          Explore |
+        </span>
+        <span className={style.collegetext}>
+          &nbsp;<span>Subjects</span>
+        </span>
+      </div>
+      <div className={style.bottomhalf}>
+        <div className={style.searchcontainer}>
+        <div className={style.img}>
+           <img src={image} alt="" />
```
Comment (codadept) [inline src/Components/Explore/Explore.jsx]: Add alt text

File: src/Components/Explore/Explore.module.scss [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+            font-weight:400;
+            font-family: 'Poppins';
+            font-style: normal;
+            color: #404040;
+            line-height: 96px;
+        }
+        .collegetext{
+            font-size: 60px;
+            font-weight:400;
+            font-family: 'Poppins';
```
Comment (codadept) [inline src/Components/Explore/Explore.module.scss]: No need to add font-family for all the elements, as its mentioned in index.scss for entire HTML elements

**Review by codadept — CHANGES_REQUESTED** [review]

File: src/Components/Explore/Explore.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import style from './Explore.module.scss';
+import image from './exploreimg.png';
+
+export default function Explore() {
```
Comment (codadept) [inline src/Components/Explore/Explore.jsx]: I have written an eslint rule to throw warning for the rule `react/function-component-definition`

Rule definition states that

> This rule is aimed to enforce consistent function types for function components. By default it prefers function declarations for named components and function expressions for unnamed components.

And by convention the function definition is preferred to be an arrow function.

So, first create an arrow function

```jsx
const Explore = () => {}
```

and then export it in the end

```jsx
export default Explore;
```

**Comment by codadept** [comment]
Few more changes required

(Following convention)
- The `Roboto` font is not imported. Import the font from [Google Fonts](https://fonts.google.com) in `index.scss` as done for Poppins font. Declare a variable naming `--gdsc-font-2` for Roboto as done for Poppins and use that variable in your Explore.module.scss.

(Following convention)
- Whatever color you are using in your scss, first check whether there is some variable in `index.scss` with that color value (most probably there will be because we are following a theme) and if not present then make a variable in `:root` in index.scss and then use that color in your scss file using `var()`

(Best practices)
- For the image `exploreimg.png`, move the image to `public/` by creating a directory such as assets and then you will be accessing the image as follows

        Suppose the image is in the following path
        `public/assets/exploreimg.png`
        
        Then you will be accessing it as 

```html
<img src="/assets/exploreimg.png" alt="Explore Image"/>
```

[Reference](https://vitejs.dev/guide/assets.html#the-public-directory)

(Following convention)
- Use the components in the following manner. As for the `Explore` Component, `src/Components/index.jsx` should contain the following lines

```jsx
import Explore from "./Explore/Explore";
import Button from "./Button/Button";
import Navbar from "./Navbar/Navbar";

export { Button, Navbar, Explore };
```

And for importing the Component in any of the file

```jsx
import { Button, Explore } from "../../Components";
```

**Comment by codadept** [comment]
image

Also increase the vertical padding of the input element a bit more to match with the design

**Comment by AnubhavChakrabortynits** [comment]
Made Changes as suggested..

Also , Increased the vertical padding of the search bar a bit... (10px - > 25px)

**Review by codadept — APPROVED** [review]

**Review by codadept — APPROVED** [review]

**Comment by codadept** [comment]
Ready to merge @AnubhavChakrabortynits ?

**Pull request merged**